ci/test/06_script_b.sh

```diff
@@ -167,7 +167,29 @@ if [ "$RUN_UNIT_TESTS" = "true" ]; then
 fi
 
 if [ "$RUN_UNIT_TESTS_SEQUENTIAL" = "true" ]; then
-  bash -c "${TEST_RUNNER_ENV} DIR_UNIT_TEST_DATA=${DIR_UNIT_TEST_DATA} LD_LIBRARY_PATH=${DEPENDS_DIR}/${HOST}/lib ${BASE_OUTDIR}/bin/test_bitcoin --catch_system_errors=no -l test_suite"
+  # The output of test_bitcoin with DEBUG_LOG_OUT is about 80MB. Thus redirect
+  # it to a file and only if errors occur, then extract the relevant snippets
+  # from it - between "Entering test case" and "Leaving test case".
+  LOG="${BASE_BUILD_DIR}/test_bitcoin.log"
+  bash -c "
+    ${TEST_RUNNER_ENV}
+    DIR_UNIT_TEST_DATA=${DIR_UNIT_TEST_DATA}
+    LD_LIBRARY_PATH=${DEPENDS_DIR}/${HOST}/lib
+    if ! ${BASE_OUTDIR}/bin/test_bitcoin --catch_system_errors=no -l test_suite -- DEBUG_LOG_OUT > ${LOG} 2>&1 ; then
+      # find all lines that match 'error: in test_suite/test_case' and extract the test suite and case
+      for t in \$(sed -E -n 's/.*error: in \"(.+\\/.+)\":.*/\\1/p' < ${LOG}) ; do
+        # t is now something like net_tests/v2transport_test
+        test_suite=\${t%/*} # e.g. net_tests
+        test_case=\${t#*/} # e.g. v2transport_test
+        ed -s ${LOG} <<EDCOMMANDS
+          /Entering test suite \"\${test_suite}\"/
+          /Entering test case \"\${test_case}\"/,/Leaving test case \"\${test_case}\"/p
+          q
+EDCOMMANDS
+      done
+      exit 1
+    fi
+  "
```

Comment:
Some of this was already there before. The new code is:

```bash
      # find all lines that match 'error: in test_suite/test_case' and extract the test suite and case
      for t in \$(sed -E -n 's/.*error: in \"(.+\\/.+)\":.*/\\1/p' < ${LOG}) ; do
        test_suite=\${t%/*} # e.g. net_tests
        test_case=\${t#*/} # e.g. v2transport_test
        ed -s ${LOG} <<EDCOMMANDS
          /Entering test suite \"\${test_suite}\"/
          /Entering test case \"\${test_case}\"/,/Leaving test case \"\${test_case}\"/p
          q
EDCOMMANDS
```

I do not think that is too different in terms of complexity from what we already have:
```make
%.cpp.test: %.cpp
        @echo Running tests: $$(\
          cat $< | \
          grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | \
          cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1\
        ) from $<
        $(AM_V_at)export TEST_LOGFILE=$(abs_builddir)/$$(\
          echo $< | grep -E -o "(wallet/test/.*\.cpp|test/.*\.cpp)" | $(SED) -e s/\.cpp/.log/ \
        ) && \
        $(TEST_BINARY) --catch_system_errors=no -l test_suite -t "$$(\
          cat $< | \
          grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | \
          cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1\
        )" -- DEBUG_LOG_OUT > "$$TEST_LOGFILE" 2>&1 || (cat "$$TEST_LOGFILE" && false)
```

If you find `ed` hard to read then I could change it to another language (python?) but I suspect it is going to be more lines of code. I any case, I don't think that's prohibitively complex task: "find 'Entering test suite', after that print the lines between 'Entering test case' and 'Leaving test case'".

Some extra escaping is needed because this is inside `bash -c "here;"`. What is the point of doing that, given that this is already a bash script? I guess doing just `here;` should have the same effect.

> I wonder how often this feature will be needed

This I do not know, but I know that when a test fails on CI that depends on a particular seed, having that seed is a game changer for fixing the bug.

Another option would be to print the seed unconditionally, even if `DEBUG_LOG_OUT` is not defined. But having the debug log in case of failure is useful beyond the rng seed.